contracts/strategies/StrategyCurveEURVoterProxy.sol

```diff
@@ -0,0 +1,215 @@
+
+    uint256 public keepCRV = 500;
+    uint256 public treasuryFee = 1000;
+    uint256 public strategistReward = 1000;
+    uint256 public withdrawalFee = 0;
+    uint256 public constant FEE_DENOMINATOR = 10000;
+
+    address public proxy;
+    address public dex;
+
+    address public governance;
+    address public controller;
+    address public strategist;
+    address public keeper;
+
+    uint256 public earned; // lifetime strategy earnings denominated in `want` token
+
+    event Harvested(uint256 wantEarned, uint256 lifetimeEarned);
+
+    constructor(address _controller) public {
+        governance = msg.sender;
+        strategist = msg.sender;
+        keeper = msg.sender;
+        controller = _controller;
+        // standardize constructor
+        proxy = address(0xC17ADf949f524213a540609c386035D7D685B16F);
+        dex = address(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
+    }
+
+    function getName() external pure returns (string memory) {
+        return "StrategyCurveEURVoterProxy";
+    }
+
+    function setStrategist(address _strategist) external {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        strategist = _strategist;
+    }
+
+    function setKeeper(address _keeper) external {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        keeper = _keeper;
+    }
+
+    function setKeepCRV(uint256 _keepCRV) external {
+        require(msg.sender == governance, "!governance");
+        keepCRV = _keepCRV;
+    }
+
+    function setWithdrawalFee(uint256 _withdrawalFee) external {
+        require(msg.sender == governance, "!governance");
+        withdrawalFee = _withdrawalFee;
+    }
+
+    function setTreasuryFee(uint256 _treasuryFee) external {
+        require(msg.sender == governance, "!governance");
+        treasuryFee = _treasuryFee;
+    }
+
+    function setStrategistReward(uint256 _strategistReward) external {
+        require(msg.sender == governance, "!governance");
+        strategistReward = _strategistReward;
+    }
+
+    function setProxy(address _proxy) external {
+        require(msg.sender == governance, "!governance");
+        proxy = _proxy;
+    }
+
+    function deposit() public {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            IERC20(want).safeTransfer(proxy, _want);
+            IVoterProxy(proxy).deposit(gauge, want);
+        }
+    }
+
+    // Controller only function for creating additional rewards from dust
+    function withdraw(IERC20 _asset) external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        require(want != address(_asset), "want");
+        require(crv != address(_asset), "crv");
+        require(weth != address(_asset), "weth");
+        require(usdc != address(_asset), "usdc");
```

Comment:
`weth` and `usdc` are only used in the path for uniswap, isn't it unnecessary to protect against the controller withdrawing these? could prevnt us from rescuing stuck funds

Reply:
In this way, any `weth` and `usdc` sitting on the strategy will all be counted as profit. If removing them, they would be treated as locked fund. Either is ok. It depends on what we want to strategy to do.

Reply:
How is that value converted to `want` though? Since `weth` and `usdc` are only used as path variables, I don't think you would benefit from any extra value locally. That value isn't converted to `want` or Curve pool shares at any point, AFAIK

Reply:
agree. removed.